File: csharp/1352.product-of-the-last-k-numbers.cs
/*
 * @lc app=leetcode id=1352 lang=csharp
 *
 * [1352] Product of the Last K Numbers
 */

using System.Collections.Generic;

// @lc code=start
public class ProductOfNumbers {

    private IList<int> Products;

    public ProductOfNumbers() {
        Products = new List<int>() { 1 };
    }
    
    public void Add(int num) {
        if (num == 0) {
            Products.Clear();
            Products.Add(1);
            return;
        }
        Products.Add(Products[Products.Count - 1] * num);
    }
    
    public int GetProduct(int k) {
        int n = Products.Count;
        if (n <= k) {
            return 0;
        }
        return Products[n - 1] / Products[n - k - 1];
    }
}

/**
 * Your ProductOfNumbers object will be instantiated and called as such:
 * ProductOfNumbers obj = new ProductOfNumbers();
 * obj.Add(num);
 * int param_2 = obj.GetProduct(k);
 */
// @lc code=end

File: java/1352.product-of-the-last-k-numbers.java
/*
 * @lc app=leetcode id=1352 lang=java
 *
 * [1352] Product of the Last K Numbers
 */

import java.util.ArrayList;
import java.util.List;

// @lc code=start

class ProductOfNumbers {

    private final List<Integer> products;

    public ProductOfNumbers() {
        products = new ArrayList<>();
        products.add(1);
    }
    
    public void add(int num) {
        if (num == 0) {
            products.clear();
            products.add(1);
            return;
        }
        products.add(products.get(products.size() - 1) * num);
    }
    
    public int getProduct(int k) {
        int n = products.size();
        if (n <= k) {
            return 0;
        }
        return products.get(n - 1) / products.get(n - k - 1);
    }
}

/**
 * Your ProductOfNumbers object will be instantiated and called as such:
 * ProductOfNumbers obj = new ProductOfNumbers();
 * obj.add(num);
 * int param_2 = obj.getProduct(k);
 */
// @lc code=end

